dali/util/shared_mem.cc

```diff
@@ -31,23 +28,31 @@ namespace python {
 ShmFdWrapper::ShmFdWrapper(int fd) : fd_{fd} {}
 
 ShmFdWrapper::ShmFdWrapper() {
-  static std::minstd_rand rand_suffix((unsigned)time(NULL) * getpid());
-  std::string name;
-  int sanity_counter = 0;
-  do {
-    std::stringstream ss;
-    ss << "/nvidia_dali_" << rand_suffix();
-    name = ss.str();
-    fd_ = shm_open(name.c_str(), O_CREAT | O_EXCL | O_RDWR, S_IRUSR | S_IWUSR);
-  } while (fd_ < 0 && errno == EEXIST && ++sanity_counter <= 5);
+  constexpr char shm_path[] = "/dev/shm/";
```

Comment:
I wonder if it is going to work on Xavier.

Reply:
You can check https://stackoverflow.com/questions/55704758/is-there-anything-like-shm-open-without-filename/55711740.
I wonder if system can create the file with the same name between call to unlink and shm_open.

Reply:
From the options listed there I believe only some variant of mkostemp applies. memfd_create which would be exactly what we need is not there in CentOS 7. It seems that call to shm_open is entirely superfluous here, but still the problem of /dev/shm vs /run/shm remains.

Reply:
Can we check if there exists /dev/shm or /run/shm? Or maybe this path is stored somewhere?